x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackClientPlugin.java

```diff
@@ -596,6 +597,11 @@
                 new ParseField(MlTasks.DATA_FRAME_ANALYTICS_TASK_NAME),
                 StartDataFrameAnalyticsAction.TaskParams::fromXContent
             ),
+            new NamedXContentRegistry.Entry(
```

Comment:
I thought the place to do this was now `MachineLearning`? XPackClientPlugin is just for the transport client that is being removed no?

The fact that we do named writeables vs named xcontent in two different places is confusing.

Reply:
Regardless, approving the PR as it fixes the issue.

Reply:
> I thought the place to do this was now `MachineLearning`?

All the other ML named x-contents related to persistent tasks were still in `XPackClientPlugin` - 3 for params and 4 for state - so this one is just adding the missing 4th one for params to the same place as the others.  Even if we decide to move all the ML ones into `MachineLearning` it will be better to do this as a separate PR rather than have one PR that moves everything and also adds a missing one.  The addition of the missing one would be disguised by the bulk move.

> The fact that we do named writeables vs named xcontent in two different places is confusing.

Yes, I don't know why #53757 moved the named writables but not the corresponding named x-contents.  The discrepancy in where they're registered is now a likely source of future bugs.